src/vm/types/signatures.rs

```diff
@@ -191,6 +218,51 @@ impl TryFrom<i128> for BufferLength {
     }
 }
 
+impl From<&StringUTF8Length> for u32 {
+    fn from(v: &StringUTF8Length) -> u32 {
+        v.0
+    }
+}
+
+impl From<StringUTF8Length> for u32 {
+    fn from(v: StringUTF8Length) -> u32 {
+        v.0
+    }
+}
+
+impl TryFrom<u32> for StringUTF8Length {
+    type Error = CheckErrors;
+    fn try_from(data: u32) -> Result<StringUTF8Length> {
+        if data * 4 > MAX_VALUE_SIZE {
+            Err(CheckErrors::ValueTooLarge)
+        } else {
+            Ok(StringUTF8Length(data))
+        }
+    }
+}
+
+impl TryFrom<usize> for StringUTF8Length {
+    type Error = CheckErrors;
+    fn try_from(data: usize) -> Result<StringUTF8Length> {
+        if data * 4 > (MAX_VALUE_SIZE as usize) {
+            Err(CheckErrors::ValueTooLarge)
+        } else {
+            Ok(StringUTF8Length(data as u32))
+        }
+    }
+}
+
+impl TryFrom<i128> for StringUTF8Length {
+    type Error = CheckErrors;
+    fn try_from(data: i128) -> Result<StringUTF8Length> {
+        if data * 4 > (MAX_VALUE_SIZE as i128) {
+            Err(CheckErrors::ValueTooLarge)
+        } else {
+            Ok(StringUTF8Length(data as u32))
```

Comment:
I think this needs to be a checked conversion -- if you add a test case like `(string-utf-8 -10)`, I think it would panic.

Reply:
Nice catch, thanks! Fixed for buff/string-ascii/string-utf8 in https://github.com/blockstack/stacks-blockchain/pull/1779/commits/3d4a57d42ecf67d2a0e2fc6502e1be8e87d994b6